### PR TITLE
pthreads4w: honor settings from profile + bump msys2

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

Fixes for Visual Studio:
- VC runtime was not honored. Upstream Makefile forces MT(d) if static, and MD(d) if shared.
- Moreover, flags build type were not injected.

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by SSE4 — APPROVED** [review]

File: recipes/pthreads4w/all/conanfile.py [hunk 1]
```diff
@@ -11,76 +13,86 @@ class Pthreads4WConan(ConanFile):
...
+                    args[0] += "-debug"
                 with tools.vcvars(self.settings):
-                    self.run('nmake %s' % ' '.join(args))
+                    with tools.environment_append(VisualStudioBuildEnvironment(self).vars):
+                        self.run("nmake {}".format(" ".join(args)))
             else:
-                self.run('autoheader', win_bash=tools.os_info.is_windows)
-                self.run('autoconf', win_bash=tools.os_info.is_windows)
+                self.run("autoheader", win_bash=tools.os_info.is_windows)
+                self.run("autoconf", win_bash=tools.os_info.is_windows)
```
[the remark this answers is not in the document]
Comment (SpaceIm) [reply recipes/pthreads4w/all/conanfile.py:78]: I was asking myself the same question 😝

File: recipes/pthreads4w/all/conanfile.py [hunk 2]
```diff
@@ -11,76 +13,86 @@ class Pthreads4WConan(ConanFile):
...
-                    args[0] += '-static'
-                if self.settings.build_type == 'Debug':
-                    args[0] += '-debug'
+                    args[0] += "-static"
+                if self.settings.build_type == "Debug":
+                    args[0] += "-debug"
                 with tools.vcvars(self.settings):
-                    self.run('nmake %s' % ' '.join(args))
+                    with tools.environment_append(VisualStudioBuildEnvironment(self).vars):
+                        self.run("nmake {}".format(" ".join(args)))
```
[the remark this answers is not in the document]
Comment (SpaceIm) [reply recipes/pthreads4w/all/conanfile.py:75]: yes, the list is overkill, can be simplified

File: recipes/pthreads4w/all/conanfile.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (madebr) [reply recipes/pthreads4w/all/conanfile.py:78]: Great minds, think ... :rofl:

File: recipes/pthreads4w/all/conanfile.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (SpaceIm) [reply recipes/pthreads4w/all/conanfile.py]: fixed

File: recipes/pthreads4w/all/conanfile.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (SpaceIm) [reply recipes/pthreads4w/all/conanfile.py]: I think you have more experience with this recipe 😉

[the remark this answers is not in the document]
Comment (madebr) [reply recipes/pthreads4w/all/conanfile.py]: See #5512.

I tested it by applying the patch to the CCI package, exporting the modified autoconf conan recipe and running autocon's test_package.

**Review by uilianries — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**